### PR TITLE
OpenJDK/OpenJFX updates

### PR DESCRIPTION
**Pull request by emilazy**
The first round of update script runs. The `openjfx17` commit is *fun*.

I have elected not to do any thorough clean‐up beyond dropping support for the now‐absent `ga` version suffixes, with the 24.11 freeze deadline in a few hours and these derivations so baroque; this is it for the upcoming release cycle, modulo version bumps. I already have plans to rewrite large portions of these packages for 25.05, so it’s not worth spending the churn and risk (albeit slight) to refactor the derivations further, when anyway superficial clean‐ups would just further expose the need for deeper ones.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/op/openjfx/package.nix [hunk 1]
```diff
@@ -69,25 +69,17 @@ stdenv.mkDerivation {
...
-        # 8338701: Provide media support for libavcodec version 61
-        # <https://github.com/openjdk/jfx23u/pull/18>
-        (fetchpatch2 {
-          url = "https://github.com/openjdk/jfx23u/commit/aba60fda1c82f00e8e685107592305c403a31287.patch?full_index=1";
-          hash = "sha256-+aRhTwi4VQthAq1SH1jxPl0mTosNMKoTY52jm+jiKso=";
-        })
-      ]
-    else if atLeast21 then
+  patches = lib.optionals (!atLeast23) (
+    if atLeast21 then
```
Comment (thiagokokada) [inline pkgs/by-name/op/openjfx/package.nix:73]: Nitpick, but I think this double version check with an inversion makes the code kinda difficult to understand.

**Review by thiagokokada — APPROVED** [review]
Nice, let's get this through the rails to get at least one version of stable with an up-to-date version of Java 🎉 .

File: pkgs/by-name/op/openjfx/package.nix (hunk 1 above)
Comment (emilazy) [reply pkgs/by-name/op/openjfx/package.nix:73]: Yeah, I’m still going back and forth on how to represent these multi‐versioned patches in these derivations. I have a prototype scheme that takes some inspiration from LLVM. The alternatives here would be `if featureVersion == "21" then … else if featureVersion == "17" then … else [ ]` or ` if atLeast21 && !atLeast23 then … else if atLeast17 then … else [ ]`, both of which are sort of awkward in their own ways. I’ll see if I can’t make it nicer for 25.05.

**Review by Infinidoge — APPROVED** [review]
Code looks good and builds just fine! Sliding in just under 24.11, hopefully!

**Comment by emilazy** [comment]
*Technically* nothing here should really be a breaking change, but I’m gonna hit the button now to avoid worrying about the technicality of `nixpkgs-update-script.openjdkSource` having slightly stricter requirements or whatever.

**Pull request merged**